### PR TITLE
Make package compatible with React 16

### PR DESCRIPTION
**Pull request by james-ecargo**
This PR updates the package to make it compatible with either React 15 or 16 by updating references to `React.createClass` and `React.propTypes`.

### CODE REVIEW

**Comment by mgechev** [comment]
@james-ecargo awesome job! Would you also add a fix for the demo?

**Comment by james-ecargo** [comment]
@mgechev sure, I'll have a look at fixing it.

**Comment by mgechev** [comment]
Thank you!

File: package.json [hunk 1]
```diff
@@ -26,9 +27,11 @@
...
   "homepage": "https://github.com/mgechev/react-reorderable",
   "dependencies": {
-    "react-drag": "^0.3.0",
-    "react": "^0.14.0",
-    "react-dom": "^0.14.2"
+    "create-react-class": "^15.0.0",
+    "prop-types": "^15.0.0",
+    "react": "^15.0.0 || ^16.0.0",
+    "react-dom": "^15.0.0 || ^16.0.0",
+    "react-drag": "^0.3.0"
```
Comment (mgechev) [inline package.json]: We can now update it to `react-drag@2.0.0`.

**Comment by mgechev** [comment]
@james-ecargo added one comment. If we have the demo working I think it's ready to merge.

**Comment by james-ecargo** [comment]
@mgechev I have fixed the demo using the production versions of React which appears to work correctly and has no warnings or errors in the browser console.

However if you use the development versions provided with the React packages:
```
diff --git a/demo/index.html b/demo/index.html
index b28edc6..619e4a6 100644
--- a/demo/index.html
+++ b/demo/index.html
@@ -48,8 +48,8 @@
 <div id="content"></div>
 <div id="events">
 </div>
-<script src="../node_modules/react/umd/react.production.min.js"></script>
-<script src="../node_modules/react-dom/umd/react-dom.production.min.js"></script>
+<script src="../node_modules/react/umd/react.development.js"></script>
+<script src="../node_modules/react-dom/umd/react-dom.development.js"></script>
 <script src="../node_modules/prop-types/prop-types.js"></script>
 <script src="../node_modules/create-react-class/create-react-class.js"></script>
 <script src="../node_modules/react-drag/dist/react-drag.js"></script>
```

Then you can observe the following error about assigning to the array of children:
```
Uncaught TypeError: Cannot assign to read only property '0' of object '[object Array]'
    at indexChildren (react-reorderable.js:112)
    at Object.getInitialState (react-reorderable.js:167)
    at new <anonymous> (create-react-class.js:870)
    at constructClassInstance (react-dom.development.js:6897)
    at updateClassComponent (react-dom.development.js:8381)
    at beginWork (react-dom.development.js:8767)
    at performUnitOfWork (react-dom.development.js:10766)
    at workLoop (react-dom.development.js:10830)
    at HTMLUnknownElement.callCallback (react-dom.development.js:580)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:619)
```
...and this one which appears to be fairly straightforward (explained [here](https://reactjs.org/docs/cross-origin-errors.html)):
```
Uncaught Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js:626)
    at invokeGuardedCallback (react-dom.development.js:476)
    at renderRoot (react-dom.development.js:10908)
    at performWorkOnRoot (react-dom.development.js:11556)
    at performWork (react-dom.development.js:11509)
    at requestWork (react-dom.development.js:11420)
    at scheduleWorkImpl (react-dom.development.js:11274)
    at scheduleWork (react-dom.development.js:11231)
    at scheduleTopLevelUpdate (react-dom.development.js:11735)
    at Object.updateContainer (react-dom.development.js:11773)
```

I am none the wiser as to why the first error happens in the development versions but doesn't appear to affect the production version.

**Review by mgechev — APPROVED** [review]

**Comment by mgechev** [comment]
@james-ecargo merged and published under react-reorderable@0.4.0.

**Pull request merged**